src/tools/illink/Directory.Build.props

```diff
@@ -0,0 +1,37 @@
+<Project>
+  <PropertyGroup>
+    <IsReferenceAssembly Condition="'$(IsReferenceAssembly)' == '' and '$([System.IO.Path]::GetFileName($(MSBuildProjectDirectory)))' == 'ref'">true</IsReferenceAssembly>
+    <DisableImplicitNamespaceImports_DotNet>true</DisableImplicitNamespaceImports_DotNet>
+    <!-- The TFM for the product (linker, task,.. not analyzer, which is netstandard) -->
+    <NetCoreAppToolCurrent>net7.0</NetCoreAppToolCurrent>
+    <!-- The TFM for all the tests - intentionally different since tests may depend on new framework APIs in order to validate
+         linker interaction with them, linker itself doesn't need the new framework typically. -->
+    <NetCoreAppTestsCurrent>net7.0</NetCoreAppTestsCurrent>
+  </PropertyGroup>
+  <PropertyGroup Condition=" '$(IsReferenceAssembly)' == 'true' ">
+    <!-- Since .NET 5 reference assemblies are always produced -->
+    <ProduceReferenceAssembly>false</ProduceReferenceAssembly>
+    <ProduceOnlyReferenceAssembly>true</ProduceOnlyReferenceAssembly>
+    <!-- Used by Arcade to compute OutputPath, IntermediateOutputPath, etc. early in the import chain. -->
+    <OutDirName>$(MSBuildProjectName)/ref</OutDirName>
+    <!-- Don't try to publish PDBs for ref assemblies that have none. -->
+    <PublishWindowsPdb>false</PublishWindowsPdb>
+    <RunApiCompat>false</RunApiCompat>
+  </PropertyGroup>
```

Comment:
Similar comment here. We already have the reference assembly infrastructure under src/libraries. Ideally we would share that instead of duplicating it.

Same applies to the Analyzers.props Import below and the ApiCompat infrastructure.